interpreter/cling/include/cling/Interpreter/Value.h

```diff
@@ -120,65 +139,67 @@ namespace cling {
     ///   dependencies.
     void AssertOnUnsupportedTypeCast() const;
 
-    size_t GetNumberOfElements() const;
+    bool isPointerType() const { return m_TypeKind == kPtrTy; }
+    bool isBuiltinType() const {
+      return m_TypeKind != kInvalid && !isPointerType();
+    };
 
-    // Allow simplisticCastAs to be partially specialized.
+    // Allow castAs to be partially specialized.
     template<typename T>
     struct CastFwd {
       static T cast(const Value& V) {
-        EStorageType storageType = V.getStorageType();
-        switch (storageType) {
-        case kSignedIntegerOrEnumerationType:
-          return (T) V.getAs<long long>();
-        case kUnsignedIntegerOrEnumerationType:
-          return (T) V.getAs<unsigned long long>();
-        case kDoubleType:
-          return (T) V.getAs<double>();
-        case kFloatType:
-          return (T) V.getAs<float>();
-        case kLongDoubleType:
-          return (T) V.getAs<long double>();
-        case kPointerType:
-        case kManagedAllocation:
+        if (V.needsManagedAllocation() || V.isPointerType())
```

Comment:
How can `V.needsManagedAllocation()` be `true` and `V.isPointerType()` be false? I.e. isn't `V.isPointerType()` enough?

Reply:
ping @vgvassilev

Reply:
Yes, you are right. I changed that.